avm/res/network/local-network-gateway/main.bicep

```
@@ -0,0 +1,157 @@
+metadata name = 'Local Network Gateways'
+metadata description = 'This module deploys a Local Network Gateway.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Local Network Gateway.')
+@minLength(1)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. List of the local (on-premises) IP address ranges.')
+param localAddressPrefixes array
+
+@description('Required. Public IP of the local gateway.')
+param localGatewayPublicIpAddress string
+
+@description('Optional. The BGP speaker\'s ASN. Not providing this value will automatically disable BGP on this Local Network Gateway resource.')
+param localAsn string = ''
+
+@description('Optional. The BGP peering address and BGP identifier of this BGP speaker. Not providing this value will automatically disable BGP on this Local Network Gateway resource.')
+param localBgpPeeringAddress string = ''
+
+@description('Optional. The weight added to routes learned from this BGP speaker. This will only take effect if both the localAsn and the localBgpPeeringAddress values are provided.')
+param localPeerWeight string = ''
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
```

```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

In alignment with the specs. Requires a regeneration of the files.